src/Symfony/Component/HttpKernel/HttpCache/Esi.php

```diff
@@ -162,8 +162,9 @@ public function process(Request $request, Response $response)
 
         // we don't use a proper XML parser here as we can have ESI tags in a plain text response
         $content = $response->getContent();
-        $content = preg_replace('#<esi\:remove>.*?</esi\:remove>#', '', $content);
-        $content = preg_replace('#<esi\:comment[^>]*(?:/|</esi\:comment)>#', '', $content);
+        $content = preg_replace('#<esi\:remove>.*?</esi\:remove>#s', '', $content);
+        $content = preg_replace('#<esi\:comment[^>]+>#s', '', $content);
+        $content = preg_replace('#<esi\:comment[^>]*>(.+?)</esi\:comment>#s', '', $content);
```

Comment:
Hihi I know :) but the original regex allowed this though, should we only support the standard `<comment text="some comment"/>` ?